battleship.py

```diff
@@ -156,6 +156,39 @@ def shipdown(y, x, ourbombboard, opshipspoints):
                 if ourbombboard[y][x] != 2:
                     return False
             return True
+
+def firstfree(ourbombboard, y, x, dx, dy):

```

Comment:
Just noticed that this should be `dy, dx`, not `dx, dy`. Was searching for why the AI was stuck at around 60 moves to win instead of 50.